### PR TITLE
fix: Fix Evaluated value paths for objects

### LINKED ISSUE #24397: [Bug]: REST API: application crashes when typing in dynamic binding in REST API query URL input box
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

- When user types in a URL with dynamic binding in REST API query URL input box then application crashes. 
- This issue was reported as part of #24218 
- [Ref](https://theappsmith.slack.com/archives/C040LHZN03V/p1686577919515999?thread_ts=1686300736.679729&cid=C040LHZN03V)

https://github.com/appsmithorg/appsmith/assets/1757421/214f87e6-7d8b-45dd-a8e4-a0309cc02326

### Steps To Reproduce

1. Create a JS Object and store a url in appsmith store say key is api_url.
2. Create new API query.
3. Edit the API query URL input box and write slowly : {{appsmith.store.api_url}}

### Public Sample App

_No response_

### Environment

Production

### Issue video log

_No response_

### Version

cloud, self hosted

### PR DESCRIPTION
**Pull request by Irongade**
This issue fixes the evaluated value for path fields that holds objects, booleans or arrays.

#### PR fixes following issue(s)
Fixes #24397 

- Bug fix (non-breaking change which fixes an issue)

- [x] Manual
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [x] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

**Review by AmanAgarwal041 — DISMISSED** [review]

**Review by AmanAgarwal041 — APPROVED** [review]

File: app/client/cypress/e2e/Regression/ServerSide/ApiTests/API_EvaluatedValue_spec.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { ObjectsRegistry } from "../../../../support/Objects/Registry";
+
+const apiPage = ObjectsRegistry.ApiPage;
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ServerSide/ApiTests/API_EvaluatedValue_spec.ts:3]: nit: Lets use objects from ObjectsCore, so object names are same thruout all specs.

**Pull request merged**